app/src/main/java/com/x8bit/bitwarden/data/platform/datasource/network/interceptor/ResponseJsonKeyTransformerInterceptor.kt

```diff
@@ -0,0 +1,97 @@
+package com.x8bit.bitwarden.data.platform.datasource.network.interceptor
+
+import kotlinx.serialization.SerializationException
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.JsonArray
+import kotlinx.serialization.json.JsonElement
+import kotlinx.serialization.json.JsonObject
+import kotlinx.serialization.json.buildJsonObject
+import okhttp3.Interceptor
+import okhttp3.Response
+import okhttp3.ResponseBody.Companion.toResponseBody
+
+/**
+ * An OkHttp interceptor that transforms the JSON response body by converting all JSON object keys
+ * to lowercase.
+ *
+ * This interceptor is useful for handling inconsistencies in JSON responses where key casing might
+ * vary.
+ */
+class ResponseJsonKeyTransformerInterceptor : Interceptor {
+
+    override fun intercept(chain: Interceptor.Chain): Response {
+        val request = chain.request()
+        val response = chain.proceed(request)
+
+        val responseBody = response.body
+            ?: return response
+
+        // Check if the response body is JSON. Return the original response if not.
+        val contentType = responseBody.contentType()
+            ?.takeIf { it.subtype.equals("json", ignoreCase = true) }
+            ?: return response
+
+        // Read the response body as a string.
+        val source = responseBody.source()
+        source.request(Long.MAX_VALUE)
+        val responseBodyString = source
+            .buffer
+            .readString(Charsets.UTF_8)
+            .takeUnless { it.isEmpty() }
```

Comment:
Can we do this to ensure the buffer is closed:
```kotlin
        val responseBodyString = source
            .buffer
            .use {
                it
                    .readString(Charsets.UTF_8)
                    .takeUnless { it.isEmpty() }
            }
            ?: return response
```

Reply:
Removed buffer usage in favor of `response.body?.string()`.